src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
   // write code here;
+  return items.reduce((grops, item) => {
+    const keyValue = String(item[key]);
+
+    if (!grops[keyValue]) {
+      grops[keyValue] = [];
+    }
+
+    grops[keyValue].push(item);
```

Comment:
Ensure the typo in the variable name is corrected here as well, from `grops` to `groups`.